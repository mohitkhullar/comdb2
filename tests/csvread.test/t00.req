create table t1 {
schema
        {
                cstring name[10]
                cstring value[10]
                cstring time[10]
        }
}$$

create procedure csv_to_table version 'def' { 
local function main()
    db:csv_open("/home/mk/data/mohitdb1/agg.csv")
    local lua_tbl = db:csv_read(",")
    print(lua_tbl)
    local tbl = db:table("t1")
    print(tbl)
    while(lua_tbl)
    do
        print(lua_tbl)
        table.foreach(lua_tbl, print)
        local x = {name = lua_tbl[1], value=lua_tbl[2], time=lua_tbl[3]}
        tbl:insert(x)
        lua_tbl = db:csv_read(",")
    end
end
}$$

put default procedure csv_to_table 'def'

exec procedure csv_to_table()

create procedure csv_to_table version 'def' { 
local function main(file)
    db:csv_open("DBDIR/agg.csv")
    local lua_tbl = db:csv_read(",")
    local tbl = db:table("t1")
    while(lua_tbl)
    do
        local x = {name = lua_tbl[1], value=lua_tbl[2], time=lua_tbl[3]}
        tbl:insert(x)
        lua_tbl = db:csv_read(",")
    end
end
}$$

exec procedure csv_to_table()

select * from t1
