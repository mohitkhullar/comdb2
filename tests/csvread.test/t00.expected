(version='def')
[exec procedure csv_to_table()] failed with rc -3 [db:csv_open("/home/mk/data/mohi...]:3: File not in database directory
(version='def')
(name='dfgjhk', value=' efsdss', time=' k23')
(name='absdcd', value=' ee1fsdss', time=' kss')
(name='dh42bcd', value=' efsfsdss', time=' k42')
(name='ab21cd', value=' ef23sdss', time=' k223')
